app/src/main/java/com/mobile/tandil/kotlinbaseproject/activity/ParkingStayActivity.kt

```diff
@@ -0,0 +1,86 @@
+package com.mobile.tandil.kotlinbaseproject.activity
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.mobile.tandil.kotlinbaseproject.databinding.ActivityParkingStayBinding
+import com.mobile.tandil.kotlinbaseproject.fragment.DatePickerDialogFragment
+import com.mobile.tandil.kotlinbaseproject.fragment.TimePickerDialogFragment
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel.CheckState
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel.ParkingStayData
+
+class ParkingStayActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityParkingStayBinding
+    private val viewModel = ParkingStayViewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityParkingStayBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.getValue().observe({ lifecycle }, ::updateUI)
+
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.startDateTimeButton.setOnClickListener {
+            viewModel.showStartTimePickerFragment()
+            viewModel.showStartDatePickerFragment()
+        }
+        binding.endDateTimeButton.setOnClickListener {
+            viewModel.showEndTimePickerFragment()
+            viewModel.showEndDatePickerFragment()
+        }
+        binding.confirmButton.setOnClickListener {
+            viewModel.closeActivity()
+        }
+        binding.cancelButton.setOnClickListener {
+            viewModel.closeActivity()
+        }
+    }
+
+    private fun updateUI(data: ParkingStayData) {
+        when (data.state) {
+            CheckState.SHOW_START_DATE_PICKER_FRAGMENT -> {
+                val datePicker = DatePickerDialogFragment { onStartDateSelected(it) }
+                datePicker.show(supportFragmentManager, DatePickerDialogFragment.DATE_PICKER_DIALOG)
+            }
+            CheckState.SHOW_START_TIME_PICKER_FRAGMENT -> {
+                val timePicker = TimePickerDialogFragment { onStartTimeSelected(it) }
+                timePicker.show(supportFragmentManager, TimePickerDialogFragment.TIME_PICKER_DIALOG)
+            }
+            CheckState.SHOW_END_DATE_PICKER_FRAGMENT -> {
+                val datePicker = DatePickerDialogFragment { onEndDateSelected(it) }
+                datePicker.show(supportFragmentManager, DatePickerDialogFragment.DATE_PICKER_DIALOG)
+            }
+            CheckState.SHOW_END_TIME_PICKER_FRAGMENT -> {
+                val timePicker = TimePickerDialogFragment { onEndTimeSelected(it) }
+                timePicker.show(supportFragmentManager, TimePickerDialogFragment.TIME_PICKER_DIALOG)
+            }
+            CheckState.CLOSE_ACTIVITY -> {
```

Comment:
if you need to close an activity just call the method finish()

Reply:
done